src/ast/query.rs

```diff
@@ -1627,6 +1627,14 @@ pub struct OrderByExpr {
     pub asc: Option<bool>,
     /// Optional `NULLS FIRST` or `NULLS LAST`
     pub nulls_first: Option<bool>,
+    /// Optional: `WITH FILL`
+    /// Supported by [ClickHouse syntax]: <https://clickhouse.com/docs/en/sql-reference/statements/select/order-by#order-by-expr-with-fill-modifier>
+    pub with_fill: Option<WithFill>,
+    /// Optional: `INTERPOLATE`
+    /// Supported by [ClickHouse syntax]
+    ///
+    /// [ClickHouse syntax]: <https://clickhouse.com/docs/en/sql-reference/statements/select/order-by#order-by-expr-with-fill-modifier>
+    pub interpolate: Option<InterpolationArg>,
```

Comment:
```suggestion
    pub interpolate: Option<Interpolation>,
```
Yeah regarding the naming, I think it would be reasonable in this case to have name the struct `Interpolate` as well so we get `pub interpolate: Interpolate`. We could have something like

```
struct Interpolate {
    exprs: Option<Vec<ExprWithAlias>>
}
```
thinking e.g.
```
INTERPOLATE => Interpolate { exprs: None}
```
```
INTERPOLATE () => Interpolate { exprs: Some(vec![]) }
```
[ExprWithAlias can be reused from here](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/ast/query.rs#L831-L843)

Reply:
I'll update the naming to `Interpolate` instead of `Interpolation`.

I don't think this can use `ExprWithAlias` since the `AS x` part is actually an expression, not an alias. For [example](https://fiddle.clickhouse.com/846d0c38-136e-4743-8256-0d2fe6297123), this is valid:

```sql
INTERPOLATE(thing AS toDecimal64(log2(thing) * 2 + 1, 3))
```

So I think we need this to be an `Expr`. After doing more digging, it looks like the initial column can only be an `Expr::Identifier(Ident)` so updating the `Interpolate.column` to be `Ident` makes the most sense (to me).

Reply:
I pushed [47ba442](https://github.com/sqlparser-rs/sqlparser-rs/pull/1330/commits/47ba442bdf6e39a5f7ae3626ca6f34ffb8986424) that swaps `column` to an `Ident` and updates the naming, etc.

Reply:
Ah that makes sense!